src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Overrides.Android.cs

```diff
@@ -56,6 +56,12 @@ public override IEditable Replace(int start, int end, ICharSequence tb, int tbst
 				copy.Replace(start, end, tb, tbstart, tbend);
 				var previewText = copy.ToString();
 
+				var coercedText = Owner.CoerceText(previewText);
+				if (coercedText == DependencyProperty.UnsetValue)
```

Comment:
I'm skeptical that this is the right thing to do, the coerce callback should be called by the DP mechanism when the Text is set, which is done by `ProcessTextInput()` a few lines below.

Let's discuss this tomorrow.

Reply:
I'm have doubt about it as well. However, it seems to be the only solution that I could found so far.
The problem is the line below: `base.Replace(...);`, which will trigger a `OnTextChanged` with a different value than `previewText`. as you can see from the note below:
```c#
@xy === BeforeTextChanging(sender, args.NewText: 12345)
    @@ call stack
    @ TextBoxStringBuilder.Replace(start: 4, end: 4, tb: "7", tbstart: 0, tbend: 1)
        var copy = new SpannableStringBuilder(this);
        copy.Replace(start, end, tb, tbstart, tbend); 
        var previewText = copy.ToString();// "(123745)"
        
        var finalText = Owner.ProcessTextInput(previewText);
        if (Owner._wasLastEditModified = previewText != finalText)
        {
            // ...
            var replacement = SOME_JAVA_STUFF(finalText); // "(12345)"
            base.Replace(0, Length(), replacement, 0, finalText.Length); // full replacement
                // in stack order
                0x2C in Windows.UI.Xaml.Controls.TextBoxView.OnTextChanged at D:\code\framework\uno\src\Uno.UI\UI\Xaml\Controls\TextBox\TextBoxView.Android.cs:83,6	C#
                0x18 in Android.Widget.TextView.n_OnTextChanged_Ljava_lang_CharSequence_III	C#
                0x29 in Android.Runtime.DynamicMethodNameCounter.123	C#
                0xFFFFFFFFFFFFFFFF in Java.Interop.NativeMethods.java_interop_jnienv_call_nonvirtual_object_method_a	C#
                0x79 in Java.Interop.JniEnvironment.InstanceMethods.CallNonvirtualObjectMethod	C#
                0x5F in Java.Interop.JniPeerMembers.JniInstanceMethods.InvokeVirtualObjectMethod	C#
                0x87 in Android.Text.SpannableStringBuilder.Replace	C#
                // reaching here the parameter `text` is already "12345"; the braces got dropped already !!!
        }
    @ TextBoxView.OnTextChanged(text: "12345", start: 0, lengthBefore: 7, lengthAfter: 5) 
        /* ^ note: the parenthesis from caller are missing
             which we have no control of, since between this call and `base.Replace(...)` from stack above is just java
             presumably java coerce this input against `InputScope=Number` and dropped any invalid character
             this and max-length constraint is what causes the flickering between "(12345)" and "12345" */
```